swift/ql/lib/codeql/swift/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -1334,10 +1344,34 @@ module Exprs {
     override InterpolatedStringLiteralExpr ast;
 
     final override ControlFlowElement getChildElement(int i) {
-      none() // TODO
+      i = 0 and
+      result.asAstNode() = ast.getAppendingExpr().getFullyConverted()
+    }
+  }
+
+  private class TapExprTree extends AstStandardPostOrderTree {
```

Comment:
I think I'm struggling to review this PR because there aren't enough comments explaining what things are supposed to be.  A good start would be explaining `TapExpr` and `TapExprTree`.

Reply:
Good point. This particular expression is very esoteric. I can add a QLDoc comment to the `TapExpr` class to explain its semantics.

Reply:
I've added a couple more comments in 6cfeb24d940b9f2a8aff2fd9c06479df235802e2. Please let me know if things still aren't clear!

Reply:
Thanks, that's a huge improvement.  I still have two questions:
 - where might they be generated?  (from context, I'm guessing an interpolated string literal will generate one?)
 - what does the 'Tap' bit stand for?

Reply:
`TapExpr` was apparently introduced into the AST specifically for interpolated strings (see [here](https://github.com/apple/swift/pull/20214/commits/5a45e3b79e030f88d7af66c7589017fb9c19693f)). Reading the [documentation](https://github.com/apple/swift/blob/main/include/swift/AST/Expr.h#L845) for it I get the feeling that it's probably used in more places, but I haven't found any other places yet.

And I have no idea what the 'Tap' bit refers to 😂. The commit that introduces it doesn't seem to spell this out explicitly.

Reply:
OK, I guess if we don't know the answer we can't explain it in the comments.  That aside, its a big improvement. :+1: